examples/python/async_streaming/client.py

```diff
@@ -0,0 +1,111 @@
+        self._call_finished = threading.Event()
+        self._consumer_future = None
+
+    def _response_watcher(
+            self,
+            response_iterator: Iterator[phone_pb2.StreamCallResponse]) -> None:
+        try:
+            for response in response_iterator:
+                if response.HasField("call_info"):
+                    self._on_call_info(response.call_info)
+                elif response.HasField("call_state"):
+                    self._on_call_state(response.call_state.state)
+        except Exception as e:
+            self._peer_responded.set()
+            raise
+
+    def _on_call_info(self, call_info: phone_pb2.CallInfo) -> None:
+        self._session_id = call_info.session_id
+        self._audio_session_link = call_info.media
+
+    def _on_call_state(self, call_state: phone_pb2.CallState.State) -> None:
+        logging.info("Call toward [%s] enters [%s] state", self._phone_number,
+                     phone_pb2.CallState.State.Name(call_state))
+        self._call_state = call_state
+        if call_state is phone_pb2.CallState.State.ACTIVE:
+            self._peer_responded.set()
+        if call_state is phone_pb2.CallState.State.ENDED:
+            self._peer_responded.set()
+            self._call_finished.set()
+
+    def call(self) -> None:
+        request = phone_pb2.StreamCallRequest()
+        request.phone_number = self._phone_number
+        response_iterator = self._stub.StreamCall(iter((request,)))
+        # Instead of consuming the response on current thread, spawn a consumption thread.
+        self._consumer_future = self._executor.submit(self._response_watcher,
+                                                      response_iterator)
+
+    def wait_peer(self) -> None:
+        logging.info("Waiting for peer to connect [%s]...", self._phone_number)
+        self._peer_responded.wait(timeout=None)
+        if self._consumer_future.done():
+            # If the future raises, forwards the exception here
+            self._consumer_future.result()
+        return self._call_state is phone_pb2.CallState.State.ACTIVE
+
+    def audio_session(self) -> None:
+        assert self._audio_session_link is not None
+        logging.info("Consuming audio resource [%s]", self._audio_session_link)
+        self._call_finished.wait(timeout=None)
+        logging.info("Audio session finished [%s]", self._audio_session_link)
+
+
+def process_call(executor: ThreadPoolExecutor, channel: grpc.Channel,
+                 phone_number: str) -> None:
+    call_maker = CallMaker(executor, channel, "1415926535")
+    call_maker.call()
+    if call_maker.wait_peer():
+        call_maker.audio_session()
+        logging.info("Call finished!")
+    else:
+        logging.info("Call failed: peer didn't answer")
+
+
+def run():
+    executor = ThreadPoolExecutor()
+    with grpc.insecure_channel("localhost:50051") as channel:
+        future = executor.submit(process_call, executor, channel, "1415926535")
```

Comment:
Is this a real number? Looks like the area code is actually the Bay Area. Probably want to use a `555-0100-XXXX` number, since that's [reserved for fictional use](https://en.wikipedia.org/wiki/555_(telephone_number)).